_posts/2022-12-06-integrating-auth0-with-react.md

```diff
@@ -0,0 +1,307 @@
+![Create Application Getting Started](/img/blog/integrating-auth0-with-react/create_application_alt.png){:width="797px" height="277px" loading="lazy"}
+
+Here we will be able to give our Auth0 Application a name, and a type, In our example, we will select **Single Page Web Applications**
+
+![Create Application Popup](/img/blog/integrating-auth0-with-react/create_application_popup.png){:width="600px" height="543px" loading="lazy"}
+
+Once the application has been created, Navigate to the Settings and these are the key credentials we will need to integrate Auth0 into our application
+
+![Application Settings](/img/blog/integrating-auth0-with-react/client_credentials.png){:width="760px" height="330px" loading="lazy"}
+
+One last step before we can begin integrating! We need to let Auth0 know where we are allowed to send users after authentication or logout, These are the **Allowed Callback URLs** and the **Allowed Logout URLs**, in addition, if you would like Auth0 to handle refreshing the users token and session persistence, we will need to configure the **Allowed Web Origins** aswell.
+
+We will set all three values here to `http://localhost:3000` for now (or whichever port you have configured your local React application runs on), In deployed applications, these values will be the fully qualified address of you applications (i.e. `https://example.com`)
+
+**Remember to Save your changes!**
+
+![Application URIs](/img/blog/integrating-auth0-with-react/application_uris.png){:width="426px" height="534px" loading="lazy"}
+
+<br />
+
+## Integrate with React
+
+> For this blog I will be working with **create-react-app** using the typescript template. The Auth0 SDK has full Typescript support in-built.
+
+Firstly, we will install the Auth0 SDK
+
+```sh
+npm install @auth0/auth0-react
+```
+
+The SDK has everything you will need to ingtegrate your application with Auth0, Under the hood it uses React Context to keep track of your users state.
+
+The easiest way to integrate Auth0 is by wrapping your entire application with the convenient `Auth0Provider`
+
+```tsx
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
+
+const root = ReactDOM.createRoot(
+  document.getElementById('root') as HTMLElement
+);
+root.render(
+  <React.StrictMode>
+    <Auth0Provider
+      domain='<Your Auth0 domain>'
+      clientId='<Your applications Client ID>'
+      redirectUri='http://localhost:3000'
+    >
+      <App />
+    </Auth0Provider>
+  </React.StrictMode>
+);
+```
+
+The `Auth0Provider` takes in two mandatory values, `domain` and `clientId`, Both of these can be found in your Settings of your Auth0 application. In addition, an optional but important parameter is the `redirectUri`, This will tell Auth0 where to send your users after authenticating successfully.
```

Comment:
```suggestion
The `Auth0Provider` takes in two mandatory values, `domain` and `clientId`, Both of these can be found in the Settings for your Auth0 application. In addition, an optional but important parameter is the `redirectUri`, This will tell Auth0 where to send your users after authenticating successfully.
```